### PR TITLE
add OSVDataSource

### PR DESCRIPTION
**Pull request by keshav-space**
- add OSV DataSource
- add test for OSV DataSource

Signed-off-by: Keshav Priyadarshi <git@keshav.space>

### CODE REVIEW

**Comment by TG1999** [comment]
Add doctsrings for all the functions and also add doctests/ unit tests for functions which have not been tested at all

File: vulntotal/datasources/osv.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
```
Comment (pombredanne) [inline vulntotal/datasources/osv.py:2]: Can you use the newest style of license header for consistency?

File: vulntotal/tests/test_osv.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
```
Comment (pombredanne) [inline vulntotal/tests/test_osv.py:2]: Same nit as above: we switch to simpler style of header

File: vulntotal/datasources/osv.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        "oss-fuzz": "OSS-Fuzz", 
+        "debian": "Debian", 
+        "hex": "Hex", 
+        "android": "Android", 
+    }
+
+
+def parse_advisory(response) -> Iterable[VendorData]:
+    if "vulns" in response:
+        for vuln in response["vulns"]:
```
Comment (pombredanne) [inline vulntotal/datasources/osv.py:79]: You can use a simpler approach and remove the line above `if "vulns" in response:` with this:
```suggestion
        for vuln in response.get("vulns") or []:
```

File: vulntotal/datasources/osv.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            aliases = []
+            affected_versions = []
+            fixed = []
+
+            if "aliases" in vuln:
```
Comment (pombredanne) [inline vulntotal/datasources/osv.py:84]: Prefer this:
`aliases.extend(vuln.get("aliases") or [])`

File: vulntotal/datasources/osv.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+                aliases.extend(vuln["aliases"])
+
+            if "id" in vuln:
```
Comment (pombredanne) [inline vulntotal/datasources/osv.py:87]: Same nit as above

File: vulntotal/datasources/osv.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+                aliases.append(vuln["id"])
+
+            if "affected" in vuln:
```
Comment (pombredanne) [inline vulntotal/datasources/osv.py:90]: Prefer a .get() in most cases to an if and [item] as suggested above

File: vulntotal/datasources/osv.py [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+                if "versions" in vuln["affected"][0]:
+                    affected_versions.extend(vuln["affected"][0]["versions"])
+
+                if vuln["affected"] and "ranges" in vuln["affected"][0]:
+                    if "events" in vuln["affected"][0]["ranges"][0]:
```
Comment (pombredanne) [inline vulntotal/datasources/osv.py:95]: Always prefer extracting a variable.

File: vulntotal/datasources/osv.py [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+                                    affected_versions.append(event["introduced"])
+                                if "fixed" in event:
+                                    fixed.append(event["fixed"])
+            yield VendorData(
+                aliases=sorted(list(set(aliases))),
+                affected_versions=sorted(list(set(affected_versions))),
+                fixed_versions=sorted(list(set(fixed))),
+            )
+
+def get_closest_nuget_package_name(query):
```
Comment (pombredanne) [inline vulntotal/datasources/osv.py:110]: How is this specific to OSV?
May be use a nuget.py for this instead

**Review by pombredanne — CHANGES_REQUESTED** [review]
Here are a few nit for your consideration

**Comment by keshav-space** [comment]
Addressed all the comments

**Review by pombredanne — APPROVED** [review]

**Review by pombredanne — APPROVED** [review]
LGTM... merging!

**Pull request merged**